plugins/ptp_operator/metrics/ptp4lParse.go

```diff
@@ -174,3 +179,30 @@ func handleHoldOverState(ptpManager *PTPEventManager,
 		}
 	}
 }
+
+func isFollowerOnly(ptp4lCfg *ptp4lconf.PTP4lConfig) bool {
+	if section, ok1 := ptp4lCfg.Sections["global"]; ok1 {
+		if value, ok2 := section["slaveOnly"]; ok2 {
+			log.Errorf("FollowerOnly scenario detected")
+			return value == "1"
+		}
+	}
+	return false
+}
+
+func followerOnlySyncState(role types.PtpPortRole, portID int, ptp4lCfg *ptp4lconf.PTP4lConfig) (outClockState ptp.SyncState) {
+	activeFollower, err := ptp4lCfg.ByRole(types.SLAVE)
+	activeFollowerPresent := err == nil
+	var listeningFollower ptp4lconf.PTPInterface
+	listeningFollower, err = ptp4lCfg.ByRole(types.LISTENING)
+	listeningFollowerPresent := err == nil
+
+	// If there is no port in FOLLOWING or SLAVE state after this transition then
+	// syncstate is holdover
+	if (!activeFollowerPresent || (activeFollower.PortID == portID && role == types.FAULTY)) &&
+		(!listeningFollowerPresent || (listeningFollower.PortID == portID && role == types.FAULTY)) {
+		return ptp.HOLDOVER
+	}
+
+	return ptp.FREERUN
```

Comment:
shouldn't this be LOCKED if at lease one of the followers is not FAULTY?

Reply:
I followed the same logic as in extractPTP4lEventState. Only the HOLDOVER sync state is acted upon in the ParsePTP4l function. The locked syncstate is set in ExtractMetrics
SetLastSyncState is always set to HOLDOVER here: https://github.com/edcdavid/cloud-event-proxy/blob/989b359b1c33697028319c3a8972e0727bb5da36/plugins/ptp_operator/metrics/ptp4lParse.go#L118